src/funk/fd_funk_rec.c

```diff
@@ -461,9 +461,24 @@ fd_funk_rec_remove( fd_funk_t *     funk,
   fd_funk_part_set_intern( fd_funk_get_partvec( funk, wksp ), rec_map, rec, FD_FUNK_PART_NULL );
   rec->flags |= FD_FUNK_REC_FLAG_ERASE;
 
+  /* At this point, the 5 most significant bytes should store data about the
+     transaction that the record was updated in. */
+
+  fd_funk_rec_set_erase_data( rec );
+
   return FD_FUNK_SUCCESS;
 }
 
+void
+fd_funk_rec_set_erase_data( fd_funk_rec_t * rec ) {
+  rec->flags |= ((rec->pair.xid->ul[0] & 0xFFFFFFFFFFUL) << (sizeof(unsigned long) * 8 - 40));
```

Comment:
hardcoding this to be xid->ul[0] feels like layering violation. have this take the value as an argument and set it in the acct_mgr